packaging/plugins/provision.sh

```diff
@@ -67,17 +58,83 @@ function wagon_create_package(){
         echo "## /packaging/$CONSTRAINTS_FILE doesn't exist"
         wagon create -s ../$PLUGIN_NAME/ -r -v -f
     fi
+
+}
+
+function generate_windows_plugin() {
+    # Print Plugin params
+    print_plugins_params &&
+    # Install common prerequisite
+    install_common_prereqs &&
+    # Install windows dependencies
+    install_windows_dependencies &&
+    # create_window_wagon
+    create_windows_wagon_package
+}
+
+function override_constraints_file() {
+    if [ ! -z "$CONSTRAINTS_FILE" ];then
+        cp -rf $CONSTRAINTS_FILE $PLUGIN_PATH/
+    fi
+}
+
+function clean_docker_resources() {
+  # This will clean all exited docker containers
+  docker rm $(docker ps -a -f status=exited -q)
+
+  # Remove Docker image
+  docker rmi $IMAGE_NAME --force
+}
+
+function get_wagon_builder_name() {
+	echo "## get wagon builder name"
+	WAGON_BUILDLER=$(basename $WAGON_BUILDLER_REPO | cut -f1 -d".")
 }
 
+function wagon_create_package(){
+
+    echo "## wagon create package"
+    # Checkout to the plugin
+    checkout_plugin
+
+    # Check if constraint file is provided or not
+    override_constraints_file
+
+    # This will generate a wagon file and dump it to the current plugin name
+    # directory, this should work for all linux image but for Redhat we need
+    # to build it locally since it needs subscription account
+    echo "## echo build wagon package using docker"
+    echo "clone $WAGON_BUILDLER_REPO"
+    git clone $WAGON_BUILDLER_REPO
+    # Get the name of the wagon builder project to navigate inside it
+    get_wagon_builder_name
```

Comment:
why not return to here?
also you can just move the impl to here, it's only used here

Reply:
Done